src/MassTransit/Mediator/Contexts/MediatorReceiveContext.cs

```diff
@@ -133,4 +133,18 @@ public FaultInfo(string messageType, string consumerType)
             public string ConsumerType { get; }
         }
     }
+
+
+    public class MediatorPublishReceiveContext<TMessage> :
```

Comment:
Why not just make set public so that it can be set on the class, instead of the extra code.

Reply:
only if "internal" :)